packages/data-models/tests/homonym.test.js

```diff
@@ -44,131 +45,542 @@ describe('homonym.test.js', () => {
+    let lemmaB = new Lemma('wordB', 'grc') // eslint-disable-line prefer-const
+    lemmaB.addFeature(new Feature(Feature.types.part, Constants.POFS_ADJECTIVE, Constants.LANG_GREEK))
+    const lexemeB = new Lexeme(
+      lemmaB,
+      [
+        infl6
+      ]
+    )
+    const homonymB = new Homonym([lexemeB])
+    const disambiguated = Homonym.disambiguate(homonymA, [homonymB])
+
+    expect(disambiguated.lexemes.length).toEqual(3)
+    expect(disambiguated.lexemes).toEqual(expect.arrayContaining([
+      lexemeA1,
+      lexemeA2,
+      expect.objectContaining({
+        lemma: expect.objectContaining({ word: lemmaB.word }),
+        inflections: [infl6],
+        disambiguated: true
+      })])
+    )
+    expect(disambiguated.isDisambiguated()).toBeTruthy()
+  })
+
+  it('7G Homonym disambiguation: case G (5)', () => {
+    /*
+    HomonymA contains 1 Lexeme with 3 inflections. HomonymB contains 1 Lexeme with 1 inflection.
+    The only difference between HomonymA.Lexeme1.lemma and HomonymB.Lexeme1.lemma is that
+    HomonymB.Lexeme1.lemma.word has an initial capital letter. HomonymB.Lexeme1.Inflection1
+    matches HomonymA.Lexeme1.Inflection1.
+    HomonymA.Lexeme1.Inflection1 has a stem and suffix whereas HomonymB.Lexeme1.Inflection1 has a stem but no suffix.
+    The resulting Homonym should contain 1 Lexeme. The lemma.word should be the same as HomonymB.lemma
+    (i.e. with initial upper case) and with the superset of all attributes from
+    HomonymA.Lexeme1 and HomonymB.Lexeme1 except only the inflection from HomonymA.Lexeme1.Inflection1
+    (i.e. the inflection that includes stem+suffix). Lexeme1 should be flagged as disambiguated.
+     */
+    let infl1 = new Inflection('stem', 'grc', 'suffix') // eslint-disable-line prefer-const
+    infl1.addFeature(new Feature(Feature.types.voice, Constants.VOICE_ACTIVE, Constants.LANG_GREEK))
+    let infl2 = new Inflection('stem2', 'grc') // eslint-disable-line prefer-const
+    infl2.addFeature(new Feature(Feature.types.voice, Constants.VOICE_PASSIVE, Constants.LANG_GREEK))
+    let infl3 = new Inflection('stem3', 'grc') // eslint-disable-line prefer-const
+    infl3.addFeature(new Feature(Feature.types.voice, Constants.VOICE_MEDIOPASSIVE, Constants.LANG_GREEK))
+    let lemmaA = new Lemma('word', 'grc') // eslint-disable-line prefer-const
+    lemmaA.addFeature(new Feature(Feature.types.part, Constants.POFS_VERB, Constants.LANG_GREEK))
+    const lexemeA = new Lexeme(
+      lemmaA,
+      [
+        infl1, infl2, infl3
+      ]
+    )
+    const homonymA = new Homonym([lexemeA])
+
+    // homonym with single lexeme, two possible inflections, disambiguator adds a new lexeme
+    let infl6 = new Inflection('stem', 'grc') // eslint-disable-line prefer-const
+    infl6.addFeature(new Feature(Feature.types.voice, Constants.VOICE_ACTIVE, Constants.LANG_GREEK))
+    let lemmaB = new Lemma('Word', 'grc') // eslint-disable-line prefer-const
```

Comment:
can we use a real Greek word for this test case? Just to be thorough? You could use Χράομαι and χράομαι

Reply:
Good idea, will add